crates/conjure-cp-lsp/src/server.rs

```diff
@@ -0,0 +1,143 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use tower_lsp::{
+    Client, LanguageServer, LspService, Server,
+    jsonrpc::{Error, Result},
+    lsp_types::{
+        ExecuteCommandOptions, ExecuteCommandParams, InitializeParams, InitializeResult,
+        InitializedParams, MessageType, ServerCapabilities, notification::Notification,
+    },
+};
+
+// use tracing_subscriber::info;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct NotifactionParams {
+    //to be fed into notifciation?
+    title: String,
+    message: String,
+    description: String,
+}
+
+enum CustomNotifciation {}
+
+impl Notification for CustomNotifciation {
+    type Params = NotifactionParams;
+
+    const METHOD: &'static str = "custom/notification";
+}
+
+#[derive(Debug)]
+struct Backend {
+    client: Client,
+}
+
+#[tower_lsp::async_trait]
+impl LanguageServer for Backend {
+    //this is the server implementation and manages the server response to client requests
+    async fn initialize(&self, _: InitializeParams) -> Result<InitializeResult> {
+        //first request of client to server
+        Ok(InitializeResult {
+            server_info: None,
+            capabilities: ServerCapabilities {
+                //
+                execute_command_provider: Some(ExecuteCommandOptions {
+                    commands: vec![String::from("custom.notification")],
+                    work_done_progress_options: Default::default(),
+                }),
+                //hover_provider: Some(HoverProviderCapability::Simple(true)),
```

Comment:
we shouldn't leave commented out code in the repo, can you please make a note of anything you might need somewhere else and remove these?